examples/thompson_sampling.py

```diff
@@ -0,0 +1,302 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Example: Gaussian Process
+=========================
+
+In this example we show how to implement Thompson sampling for Gaussian processes.
```

Comment:
please make it clear you're doing minimization

Reply:
I made it clear. But maybe I should add even more documentation and a plot to illustrate the idea of Bayesian optimization and Thompson sampling, right?

Reply:
i think you can keep it pretty simple. you can always link to papers/tutorials/etc for more info